base/compiler/abstractlattice.jl

```diff
@@ -0,0 +1,173 @@
+abstract type AbstractLattice; end
+function widen end
```

Comment:
It feels a little odd to merge this with the existing function of this name
```
help?> widen
  widen(x)

  If x is a type, return a "larger" type, defined so that arithmetic operations + and - are guaranteed not to overflow nor lose precision for any combination of
  values that type x can hold.
```

Reply:
I can do `widenlattice` instead.